pypose/module/liesplinese3.py

```diff
@@ -0,0 +1,53 @@
+import torch
+from torch import nn
+
+class lieSpline(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+        self.B = torch.tensor([[5, 3, -3, 1],
+                               [1, 3, 3, -2],
+                               [0, 0, 0, 1]]) / 6
+
+    def compute_weights(self, time):
+        alpha = torch.arange(4)
+        tt = time[:, None, :] ** alpha[None, :, None]
+        return self.B @ tt
+
+    def interpolateSE3(self, input_poses, time, inter_begin_and_end = True):
+        """ B spline interpolation in SE3
+        Args:
+            input_poses: input n(n>=4) poses with [batch_size, n, 4, 4] shape
+            time: k interpolation time point with [1, 1, k] shape
+        Returns:
+            output_poses: output (n-3)*k poses with [batch_size, (n-3)*k, 4, 4] shape
```

Comment:
Explain `inter_begin_and_end`

Reply:
add brief math explanation in docstring

Reply:
Have explained the parameter inter_begin_and_end and added math explanation.